test/stateful_dataloader/test_dataloader.py

```diff
@@ -1947,6 +1960,116 @@ def test_sampler_reproducibility(self):
                     ls[i].append(next(its[i]))
             self.assertEqual(ls[0], ls[1])
 
+    def test_initialization_StatefulDistributedSampler(self):
+        from torchdata.stateful_dataloader.sampler import StatefulDistributedSampler
+
+        dataset = self.dataset
+        sampler = StatefulDistributedSampler(dataset, num_replicas=10, rank=0, shuffle=False, seed=42, drop_last=False)
```

Comment:
For testing state_dict, let's have most of the tests set up with passing sampler + dataset to StatefulDataLoader so we can test that it works end-to-end

Reply:
You might need to use a dummy Collate function to easily inspect elements, check the `test_state_dict.py` file for examples

Reply:
New tests here: https://github.com/pytorch/data/blob/cdc5d318bab9432383e808dd25489677dd36edf9/test/stateful_dataloader/test_sampler.py#L173